docs/proposals/Certificate Revocation Lists.md

```diff
@@ -0,0 +1,188 @@
+# CRL and CRL Cache Design
+
+## Intro
+
+Certificate validation is an essential step during signature validation. Currently Ratify supports checking for revoked certificates through OCSP supported by notation-go library. However, OCSP validation requires internet connection for each validation while CRL could be cached for better performance. As notary-project is adding the CRL support for notation signature validation, Ratify could utilize it.
+
+## Goals
+
+CRL support, including CRL downloading, validation, and revocation list checks.
+
+- Define a cache provider interface for CRL
+- Implement default file-based cache implementation for both CLI and K8S
```

Comment:
Hi @junczhu , were there other cache types considered? could we elaborate on why file based cache is chosen? Other types of cache we use captured at https://ratify.dev/docs/next/reference/cache

Reply:
thanks for the details. Feel free to resolve this comment, after adding this info to the design doc

Reply:
Sure.